### PR TITLE
Extract interfaces to their own repos

### PR DESCRIPTION
**Pull request by lencioni**
It turns out that optionalDependencies doesn't work the way I expected
it to. It will install all of the optionalDependencies unless
`--no-optional` is used all the time by consumers when running `npm
install`. This is no good because people not using react-native or
aphrodite shouldn't have to install those dependencies.

So, I extracted the interfaces to their own repos, published with their
own packages.

- https://github.com/airbnb/react-with-styles-interface-aphrodite
- https://github.com/airbnb/react-with-styles-interface-react-native

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ Register your default theme and interface. For example, if your default theme is
 
 ```js
 import ThemedStyleSheet from 'react-with-styles/lib/ThemedStyleSheet';
-import aphroditeInterface from 'react-with-styles/lib/interfaces/aphroditeInterface';
+import aphroditeInterface from 'react-with-styles-interface-aphrodite';
```
Comment (ljharb) [inline README.md]: should we link to these repos somewhere in the readme?

Comment (lencioni) [inline README.md]: Definitely. Any suggestions? Should I add a section towards the top?

Comment (lencioni) [inline README.md]: d3c2938

**Comment by ljharb** [comment]
LGTM

**Pull request merged**